cloud/interfaces.go

```diff
@@ -54,21 +57,37 @@ type CredentialGetter interface {
 
 // Authorizer is an interface which can get the subscription ID, base URI, and authorizer for an Azure service.
 type Authorizer interface {
-	SubscriptionID() string
 	BaseURI() string
 	Authorizer() autorest.Authorizer
 }
 
-// ClusterDescriber is an interface which can get common Azure Cluster information
+// ClusterDescriber describe a Kubernetes resource with full
+// serializability, object meta and type meta, plus can describe enough
+// data to build an Azure cluster
 type ClusterDescriber interface {
```

Comment:
@CecileRobertMichon any feelings about this? how do you see scope evolving long term for a use case like this?